build-tools/src/main/kotlin/org/jetbrains/kotlin/ExecutorService.kt

```diff
@@ -311,3 +320,233 @@ private fun sshExecutor(project: Project) : ExecutorService = object : ExecutorS
         }
     }
 }
+
+private fun deviceLauncher(project: Project) = object : ExecutorService {
+    private val xcProject = Paths.get(project.testOutputRoot, "launcher")
+
+    private val idb = project.findProperty("idb_path") as? String ?: "idb"
+
+    private val deviceName = project.findProperty("device_name") as? String
+
+    override fun execute(action: Action<in ExecSpec>): ExecResult? {
+        val udid = targetUDID()
+        println("Found device UDID: $udid")
+        install(udid, "build/KonanTestLauncher.ipa")
+        val commands = startDebugServer(udid, "org.jetbrains.kotlin.KonanTestLauncher")
+                .split("\n")
+                .filter { it.isNotBlank() }
+                .flatMap { listOf("-o", it) }
+
+        var savedOut: OutputStream? = null
+        val out = ByteArrayOutputStream()
+        val result = project.exec { execSpec: ExecSpec ->
+            action.execute(execSpec)
+            execSpec.executable = "lldb"
+            execSpec.args = commands + "-b" + "-o" + "command script import ${pythonScript()}" +
+                    "-o" + ("process launch" +
+                        (execSpec.args.takeUnless { it.isEmpty() }
+                                ?.let { " -- ${it.joinToString(" ")}" }
+                                ?: "")) +
+                    "-o" + "get_exit_code" +
+                    "-k" + "get_exit_code" +
+                    "-k" + "exit -1"
+            savedOut = execSpec.standardOutput
```

Comment:
Am I understand right that we save `execSpec.standardOutput` because a user may set it in the `action`? If yes, may be add some comment about it?

Reply:
Yes. I'll add a comment here as it's a really tricky part